extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js

```diff
@@ -65,14 +66,22 @@ export class AmpAdXOriginIframeHandler {
 
     /** @private {?Promise} */
     this.adResponsePromise_ = null;
+
+    if (this.baseInstance_.emitLifecycleEvent &&
```

Comment:
I'm not convinced this saves you anything given you already check this again below for the load listener case.  Might as well just check elsewhere and save some memory allocation (event name string)